### DISCUSSION TITLE
How to handle sensor query interval

### DISCUSSION THREAD

**Discussion by cyrilpawelko** [arendst/Tasmota#10192] [Q&A] [answered]
Hello, 
I just pushed (https://github.com/arendst/Tasmota/pull/10190) support for TOF10120, which is a time of flight sensor I'm going to use for a water tank.
I copied logic from other driver sensors and hooked on FUNC_EVERY_250_MSECOND, so measurements are done 4 times per second.
This is way too much for my usage, since a 5 minutes interval (like TelePeriod) would be enough for me.
What would be the best way to achieve this, while keeping this 250ms interval for other users ?
I'm not familiar with Tasmota drive development, so any advice would be welcome.

**Comment by Jason2866** [comment]
Individual sensor query intervals are not possible. The query is hardcoded (like you did) with some fixed intervalls. It is no problem to read the sensors more often than needed. How often it is needed depends on use case. So it is better to have it more often.

**Comment by sfromis** [comment, marked as answer]
With most sensors there is no "cost" in polling the values frequently, but the PMS5003 driver is an example of prolonging effective lifetime by letting the PM sensor be passive most of the time, coded in this specific [driver](https://github.com/arendst/Tasmota/blob/development/tasmota/xsns_18_pms5003.ino#L199-L235).

**Reply by cyrilpawelko** [reply]
Thanks @Jason2866 and @sfromis for your replies.
I couldn't find information about this device lifetime, so I'll try to go like this.
I'm still waiting for my pull request #10190 to be accepted, and maybe will try to add options for data handling.
